test/misc.jl

```diff
@@ -0,0 +1,34 @@
+@testset "pixel_shuffle" begin
+    x = reshape(1:16, (2, 2, 4, 1))
+    # [:, :, 1, 1] =
+    #     1  3
+    #     2  4
+    # [:, :, 2, 1] =
+    #     5  7
+    #     6  8
+    # [:, :, 3, 1] =
+    #     9  11
+    #     10  12
+    # [:, :, 4, 1] =
+    #     13  15
+    #     14  16
+
+    y_true = [1  9 3 11
+              5 13 7 15
+              2 10 4 12
+              6 14 8 16][:,:,:,:]
```

Comment:
Based on Fig. 1 in the paper, I would expect this to be transposed.

Reply:
to be safe I checked what pytorch does:
```python
In [7]: torch.arange(1,17).reshape((1,4,2,2))
Out[7]: 
tensor([[[[ 1,  2],
          [ 3,  4]],

         [[ 5,  6],
          [ 7,  8]],

         [[ 9, 10],
          [11, 12]],

         [[13, 14],
          [15, 16]]]])

In [19]: nn.PixelShuffle(2)(x)
Out[19]: 
tensor([[[[ 1,  5,  2,  6],
              [ 9, 13, 10, 14],
              [ 3,  7,  4,  8],
              [11, 15, 12, 16]]]])
```
So current implementation seems correct, it gives the transpose of pytorch's output.
Or let me rephase it like this: inputs that are the same when flattened correspond to outputs that are the same when flattened.

Reply:
ouch, on a larger example I get different results, see below